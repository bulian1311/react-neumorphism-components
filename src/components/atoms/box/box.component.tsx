import React from 'react';
import cn from 'classnames';
import { Element } from '../../core/element';
import styles from './box.module.css';

export const Box: React.FC = ({ children }) => {
  return (
    <Element renderAs="div" className={cn(styles.box)}>
      {children}
    </Element>
  );
};

Box.displayName = 'Box';

Box.propTypes = {};

Box.defaultProps = {};
