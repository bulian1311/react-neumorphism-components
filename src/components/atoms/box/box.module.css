.box {
  display: block;
  padding: 1.25rem;
}

.box:not(:last-child) {
  margin-bottom: 1.5rem;
}
