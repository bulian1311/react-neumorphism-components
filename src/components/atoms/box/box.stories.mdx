import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as stories from './box.stories.tsx';
import { Box } from './box.component.tsx';

<Meta title="Atoms/Box" />

# Box

A box element to contain other elements

<ArgsTable of={Box} />

## Example

<Canvas>
  <Story story={stories.Box} name="Box" />
</Canvas>
