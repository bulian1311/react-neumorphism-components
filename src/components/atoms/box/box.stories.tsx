import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Box as BoxComponent } from './box.component';

export default {
  title: 'Atoms/Box',
  component: BoxComponent,
  argTypes: {},
} as Meta;

const Template: Story = () => <BoxComponent>I'm in a box.</BoxComponent>;

export const Box = Template.bind({});
Box.args = {};
