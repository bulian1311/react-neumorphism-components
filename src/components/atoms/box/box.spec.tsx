import React from 'react';
import { Box } from './box.component';
import { render, RenderResult } from '@testing-library/react';

let documentBody: RenderResult;

describe('Block', () => {
  beforeEach(() => {
    documentBody = render(<Box>Box</Box>);
  });

  it('Should Exist', () => {
    expect(Box).toBeDefined();
  });

  it('To be in the document.', () => {
    const { getByText } = documentBody;
    expect(getByText('Box')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = documentBody;
    expect(container.firstChild).toMatchSnapshot();
  });
});
