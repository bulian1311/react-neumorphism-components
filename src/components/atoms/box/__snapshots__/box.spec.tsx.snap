// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<div
  class="sc-jRQBWg NoBAS"
  shape="flat"
  size="150"
>
  Box
</div>
`;
